lib/pghero/methods/tables.rb

```diff
@@ -1,67 +1,294 @@
+                      heap_blks_hit AS hit
+                    FROM
+                      pg_statio_user_tables
+                    WHERE
+                      relid = '%s'::regclass::oid $$ )).result::bigint AS heap_blks_hit,
+                  (run_command_on_placements(logicalrelid, $$
+                    SELECT
+                      heap_blks_read
+                    FROM
+                      pg_statio_user_tables
+                    WHERE
+                      relid = '%s'::regclass::oid $$ )).result::bigint AS heap_blks_read
+                FROM
+                  pg_dist_partition
+              )
+              SELECT
+                n.nspname AS schemaname,
+                c.relname,
+                sum(p.heap_blks_hit) AS heap_blks_hit,
+                sum(p.heap_blks_read) AS heap_blks_read
+              FROM
+                placements_heap_blks_stats p
+                JOIN pg_class c ON p.logicalrelid::oid = c.oid
+                LEFT JOIN pg_namespace n ON n.oid = c.relnamespace
+              GROUP BY
+                2, 1
+            )
+            SELECT
+              schemaname AS schema,
+              relname AS table,
+              CASE WHEN heap_blks_hit + heap_blks_read = 0 THEN
+                0
+              ELSE
+                ROUND(1.0 * heap_blks_hit / (heap_blks_hit + heap_blks_read), 2)
+              END AS hit_rate
+            FROM (
+              SELECT
+                *
+              FROM
+                dist_heap_blks_stats
+              UNION ALL
+              SELECT
+                schemaname, relname, heap_blks_hit, heap_blks_read
+              FROM
+                pg_statio_user_tables s
+              WHERE
+                NOT EXISTS(SELECT * FROM dist_heap_blks_stats WHERE relname = s.relname)
+            ) all_stats
+            ORDER BY
+              2 DESC, 1
+          SQL
+        end
       end
 
       def unused_tables
-        select_all <<-SQL
-          SELECT
-            schemaname AS schema,
-            relname AS table,
-            n_live_tup AS estimated_rows
-          FROM
-            pg_stat_user_tables
-          WHERE
-            idx_scan = 0
-          ORDER BY
-            n_live_tup DESC,
-            relname ASC
-         SQL
+        if !citus_enabled?
+          select_all <<-SQL
+            SELECT
+              schemaname AS schema,
+              relname AS table,
+              n_live_tup AS estimated_rows
+            FROM
+              pg_stat_user_tables
+            WHERE
+              idx_scan = 0
+            ORDER BY
+              n_live_tup DESC,
+              relname ASC
+           SQL
+        else
+          select_all <<-SQL
+            WITH dist_tables_stats AS (
+              WITH dist_tables_with_indexes (tablename) AS (
+                SELECT
+                  indrelid::regclass
+                FROM
+                  pg_index
+                  JOIN pg_dist_partition ON indrelid = logicalrelid
```

Comment:
This query repeats same distributed tables multiple times if they have multiple indexes, is this what we want? If not, consider using DISTINCT.

Reply:
That's a very good point 👍 Will use DISTINCT